### ISSUE TITLE
Show both name and state text in a button?

### ISSUE THREAD

**Issue by crowbarz** [jcwillox/lovelace-paper-buttons-row#3]
Is it possible to show both name and state text (of either the button's entity, or ideally the state of a different entity) as part of a button? I'm trying to create a button representing an outlet that displays its name above an icon which shows the outlet state via colour and toggles the outlet's switch when clicked, as well as show the outlet's power usage (captured in the state of a template sensor exposing an attribute of the outlet) underneath the icon.

I have only been able to use the `state_text` map to show static text on the button, but this text replaces the name, and also this mechanism isn't ideal for numeric sensors that don't have a limited set of discrete values that one can create a map for.

I can almost do this with a glance card button and a custom `tap_action` to implement the toggle, except I can't get the icon colour to reflect the state because the button's underlying entity needs to be the power usage sensor in order to show the power usage underneath the icon. So I was hoping maybe to be able to implement it through this plugin.

**Comment by jcwillox** [comment]
It sounds like what your looking for is some kind of templating. I currently don't have any plans to include templating within this plugin, I may look into this. For now, try to find some kind of card templater or for templated styling try `card-mod`.

**Comment by crowbarz** [comment]
I was maybe hoping that the `state_text` option could be overloaded to accept `true` to display the state of the underlying entity as-is (optionally including `unit_of_measurement`) as well as accepting an entity-id to display the state of that entity instead. Templating would work here, though I would suggest that at least showing the entity's state as-is rather than applying through a map is something that there are probably other use cases for (eg. when using any numeric sensor like a temperature gauge or a counter).

Plus it would be nice to be able to display the entity's name separately to the state value, on the opposite side of the icon (if present) to the state as defined by `align_icon`.

Thus this hypothetical configuration (sans end-of-line comments - is that valid YAML??):
```
  - buttons:
      - align_icon: top # <- icon is above the state text
        entity: switch.horizon_outlet_switch
        icon: 'mdi:power-socket-uk'
        name: Horizon Outlet # <- this text would go on the opposite side of the icon to the state
        state_text: sensor.horizon_outlet_power # <- this shows the state of another entity
        tap_action:
          action: more-info
        hold_action:
          action: toggle
        state_styles: # emulate state_color: true
          'off':
            icon:
              color: var(--paper-item-icon-color)
          'on':
            icon:
              color: var(--paper-item-icon-active-color)
        style: # don't change text colour along with state
          name: # <- hypothetically be able to style the name separately to the state ("text")
            color: var(--primary-text-color)
            padding-bottom: 10px
          text:
            color: var(--primary-text-color)
            padding-bottom: 10px
    type: 'custom:paper-buttons-row'
```
would generate something like this:
```
+-----------------------+
|     Horizon Outlet    |
|        [ Icon ]       | <- colour changes with state
|         40.7 W        |
+-----------------------+
```

**Comment by dimmanramone** [comment]
@jcwillox I think that @crowbarz is after something like the glance card that can change the icon color based on the state. If I'm not mistaken the only way to do that with glance card is to use the the [custom ui component](https://github.com/andrey-git/home-assistant-custom-ui) (at least that is the way I have that in my configuration). I think that is great idea to have the option to change the just the icon color and show both entity name (or just a variable name) and state of the entity at the same time.

**Comment by jcwillox** [comment]
In `paper-buttons-row` you are able to change the icon colour per-state.

Thanks for that example image that does look like what he wants, and has given me a better idea on how to implement that, likely with a `show_state` option. 

I'll still need to find a way to implement it hopefully without breaking changes and without making the configuration more complicated, but ill look into implementing it when I next get a chance.

Sidenote I've written a plugin (not released) that allows you to change icon colour based on either specific states or thresholds, so you could change the colour of a battery entity as it drains, something like that may be more applicable.

**Comment by Shaun-Harrison** [comment]
Hello

Also looking for this! 

Working on a card that will play the latest home assistant podcasts , used scrape to get the mp3 url and titles

![image](https://user-images.githubusercontent.com/46562010/80080545-ad034180-8549-11ea-9ac7-b2d8f13dda6b.png)

Currently using entity card to display them, but forced to have 1-5 in the names 
![image](https://user-images.githubusercontent.com/46562010/80080655-d1f7b480-8549-11ea-9d5c-95b86cafd0bc.png)

Would be great if we could use something like 
    - type: custom:paper-buttons-row
      buttons:
        - icon: mdi:home-import-outline
          name: '{{ states.sensor.podcast_homeassistant_1_title.state }}'

So that the name is the sensor state

**Comment by RichmonS** [comment]
Was this ever fixed/addressed?  I am using this custom button so I can have different icons than what HA uses for the default domain.  The issue is that the button is centered on the card.  I would like the icon and entity name to be left justified and the state right justified.  Below is the current YAML for my card.


type: entities
title: Main Floor
entities:
  - type: 'custom:paper-buttons-row'
    buttons:
      - entity: binary_sensor.wyzesense_779c204b
        state_icons:
          'off': 'mdi:window-closed'
          'on': 'mdi:window-opened'
        state_text:
          'off': Office Window Left Closed
          'on': Office Window Left Open
        style:
          text-align: left
  - type: 'custom:paper-buttons-row'
    buttons:
      - entity: binary_sensor.wyzesense_779c2079
        state_icons:
          'off': 'mdi:window-closed'
          'on': 'mdi:window-opened'
        state_text:
          'off': Office Window Right Closed
          'on': Office Window Right Open

**Comment by jcwillox** [comment]
Well, there's technically two features talked about here the "introduction of templates" and "add the literal state as text" neither of which has been added yet (otherwise this issue would be closed).
I'm quite busy at the moment so ill try to address these when I next get the chance.

Seems like your talking about having a row like this but with the state included?
![image](https://user-images.githubusercontent.com/39120423/81242186-d21aa880-904f-11ea-94aa-dd58d90676df.png)
```yaml
style:
  button:
    background-color: var(--table-row-background-color)
    border: 1px solid
    border-radius: 4px
    width: 100%
    color: '#4dd0e1'
  text:
    padding-left: 20px
```

**Comment by RichmonS** [comment]
That is correct.  Using the picture above it would be instead of the switch it would be the state.  I am trying to change the icon for an open sensor to a window icon instead of the default door.  So instead of the light bulb there would be my icons based on state, then the name of the entity left justified.  Where the switch is location have the state right justified.  This would be what the current entities card would look like.

**Comment by ktownsend-personal** [comment]
I would like to add my voice to wanting to show state (and unit). I just found this element today and it's just nearly a perfect fit for my picture elements card for managing my pool system. This works great for many of my control buttons, but some of them need to show temperature. 

I love the way this element works, and the flexibility of the styling and state value substitutions. Before using paper-buttons-row I had to resort to separate icon and state elements, which are impossible to place together as a visual pair when using relative placement and you are showing them on different size screens. Other elements I tried just haven't achieved the right features or don't look good because they were designed to use in an entities card and try to preserve the column alignments (large gaps between icon and text for example).
